src/dd-win-prof/StackSamplerLoop.cpp

```diff
@@ -266,19 +281,37 @@ void StackSamplerLoop::CollectOneThreadSample(std::shared_ptr<ThreadInfo>& pThre
             _pCpuTimeProvider->Add(std::move(sample), duration);
         }
         else
-            if (profilingType == PROFILING_TYPE::WallTime)
-            {
-                Sample sample = Sample(
-                    thisSampleTimestamp,
-                    pThreadInfo,
-                    frames,
-                    framesCount);
-                _pWallTimeProvider->Add(std::move(sample), duration);
-            }
-            else
+        if (profilingType == PROFILING_TYPE::WallTime)
+        {
+            std::chrono::nanoseconds waitDuration = 0ns;
+
+            // check if the thread is waiting
+            if (waitingReason != WAIT_REASON_NONE)
             {
-                // should neven happen
+                // compute the "current" wait duration
+                // since we don't have the start/ end time of the wait, we "jump" from wait to wait
+                auto lastWaitTimestamp = pThreadInfo->SetLastWaitSampleTimestamp(thisSampleTimestamp);
+                if (lastWaitTimestamp != 0ns)
+                {
+                    waitDuration = thisSampleTimestamp - lastWaitTimestamp;
```

Comment:
Interesting proposal.
We will have to experiment with this.
So the logic is currently
- Store the last waiting sample
- Compute the wait time based on last waiting sample
- Reset the wait when we are on CPU